src/Services/ReferenceTokenValidator.cs

```diff
@@ -0,0 +1,51 @@
+using System.Security.Cryptography;
+using System.Text;
+using Duende.IdentityServer.EntityFramework.DbContexts;
+using IdentityServerConfig.Models;
+
+namespace IdentityServerConfig.Services;
+
+public class ReferenceTokenValidator : IReferenceTokenValidator
+{
+    public bool IsActive(DateTime? expirationDate)
+    {
+        return expirationDate == null || expirationDate >= DateTime.UtcNow;
+    }
+
+    public ReferenceTokenDataModel Validate(CheckReferenceTokenModel checkReferenceTokenModel,
+        PersistedGrantDbContext persistedGrantDbContext)
```

Comment:
This can probably be injected through the constructor rather than having to provide it every time we call this method.

Reply:
Fixed it in 03775a1